File: app/jms/token.py
from wisp.protobuf import service_pb2
from wisp.exceptions import WispError
from wisp.protobuf.common_pb2 import TokenAuthInfo

from jms.base import BaseWisp
from utils.logger import get_logger

logger = get_logger(__name__)


class TokenHandler(BaseWisp):

    def get_token_auth_info(self, token: str) -> TokenAuthInfo:
        req = service_pb2.TokenRequest(token=token)
        token_resp = self.stub.GetTokenAuthInfo(req)
        if not token_resp.status.ok:
            error_message = f'Failed to get token: {token_resp.status.err}'
            logger.error(error_message)
            raise WispError(error_message)
        return token_resp.data

File: pkg/jms/token.go
package jms

import (
	"context"
	"errors"
	"github.com/jumpserver/kael/pkg/httpd/grpc"
	"github.com/jumpserver/kael/pkg/logger"
	"github.com/jumpserver/wisp/protobuf-go/protobuf"
)

type TokenHandler struct{}

func NewTokenHandler() *TokenHandler {
	return &TokenHandler{}
}

func (th *TokenHandler) GetTokenAuthInfo(token string) (*protobuf.TokenAuthInfo, error) {
	ctx := context.Background()
	req := &protobuf.TokenRequest{Token: token}

	resp, _ := grpc.GlobalGrpcClient.Client.GetTokenAuthInfo(ctx, req)
	if !resp.Status.Ok {
		errorMessage := "Failed to get token: " + resp.Status.Err
		logger.GlobalLogger.Error(errorMessage)
		return nil, errors.New(errorMessage)
	}
	return resp.Data, nil
}
